slsim/lens.py

```diff
@@ -1359,6 +1367,26 @@ def generate_id(self, ra=None, dec=None):
 
         return f"{lens_type}-LENS_{ra:.4f}_{dec:.4f}"
 
+    def add_subhalos(
```

Comment:
please add a test function, also notice that you need pyHalo as a requirement (or at least a test_requirement)